src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -26,5 +28,44 @@ private static bool GetGlobalizationInvariantMode()
             }
             return invariantEnabled;
         }
+
+        private static bool LoadIcu()
+        {
+            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            {
+                return Interop.Globalization.LoadICU() != 0;
+            }
+
+#if TARGET_OSX
+            string extension = ".dylib";
+            bool versionAtEnd = false;
+#else
+            string extension = "so.";
+            bool versionAtEnd = true;
+#endif
+
+            // Append '.' to suffix since in Unix the version is separated by '.'
+            int suffixLength = suffix.Length + 1;
+            Span<char> suffixWithSeparator = stackalloc char[suffixLength];
+            suffix.CopyTo(suffixWithSeparator);
+            suffixWithSeparator[suffixLength - 1] = '.';
+
+            Assembly assembly = Assembly.GetExecutingAssembly();
```

Comment:
`GetExecutingAssembly` is not a cheap API. Use e.g. `typeof(object).Assembly` instead.

Reply:
Also, you do not need worry about caching it. It is ok to call `typeof(object).Assembly` a few extra times here.